test_functional.py

```diff
@@ -0,0 +1,106 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+import pytest
+import socket
+#import server
+from multiprocessing import Process
+#from server import response_ok
+
+
+def run_server():
+    ADDR = ('127.0.0.1', 8000)
+    socket_conn = socket.socket(
+        socket.AF_INET, socket.SOCK_STREAM, socket.IPPROTO_IP
+    )
+
+    socket_conn.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+
+    socket_conn.bind(ADDR)
+
+    socket_conn.listen(1)
+
+    while True:
+        try:
+            conn, addr = socket_conn.accept()
+            output = ""
+            while True:
+                msg = conn.recv(16)
+                output = output + msg
+                if len(msg) < 16:
+                    #response_ok()
+                    conn.sendall(output)
+                    conn.close()
+                    break
+        except KeyboardInterrupt:
+            break
+
+
+@pytest.yield_fixture()
+def multiproc():
+    special_process = Process(target=run_server)
+    special_process.daemon = True
+    special_process.start()
+    yield special_process
+
+
+def test_response_ok(multiproc):
+    pass
```

Comment:
It seems we aren't quite clear on what response_ok is supposed to do yet.  same for response_error below.